state/pgstatestorage/forkid.go

```diff
@@ -81,9 +81,11 @@ func (p *PostgresStorage) AddForkIDInterval(ctx context.Context, newForkID state
 		var forkIDs []state.ForkIDInterval
 		forkIDs = oldForkIDs
 		// Check to detect forkID inconsistencies
-		if forkIDs[len(forkIDs)-1].ForkId+1 != newForkID.ForkId {
-			log.Errorf("error checking forkID sequence. Last ForkID stored: %d. New ForkID received: %d", forkIDs[len(forkIDs)-1].ForkId, newForkID.ForkId)
-			return fmt.Errorf("error checking forkID sequence. Last ForkID stored: %d. New ForkID received: %d", forkIDs[len(forkIDs)-1].ForkId, newForkID.ForkId)
+		if forkIDs[len(forkIDs)-1].ForkId >= newForkID.ForkId {
```

Comment:
While this fixes "my" problem, I believe that this restriction shouldn't be here. Conceptually, the node is an implementation of the protocol, and the protocol allow to move from ForkID X to Y, where Y < X, and even X == Y.

FYI @invocamanman